hibernate-core/src/main/java/org/hibernate/boot/query/HbmResultSetMappingDescriptor.java

```diff
@@ -257,7 +260,22 @@ public NamedResultSetMappingMemento resolve(ResultSetMappingResolutionContext re
 				(descriptor) -> resultMementos.add( descriptor.resolve( resolutionContext ) )
 		);
 
-		return new NamedResultSetMappingMementoImpl( registrationName, resultMementos );
+		List<DynamicFetchBuilderLegacy> fetchBuilderLegacies = new ArrayList<>();
+		for ( Map.Entry<String, Map<String, JoinDescriptor>> entry : joinDescriptors.entrySet() ) {
+			final Map<String, JoinDescriptor> value = entry.getValue();
+			String ownerTableAlias = entry.getKey();
+			for ( Map.Entry<String, JoinDescriptor> v : value.entrySet() ) {
+				final JoinDescriptor descriptor = v.getValue();
+				final DynamicFetchBuilderLegacy fetchBuilder = Builders.fetch(
+						descriptor.tableAlias,
+						ownerTableAlias,
+						descriptor.propertyPath
+				);
+				fetchBuilderLegacies.add( fetchBuilder );
+			}
+		}
+
+		return new NamedResultSetMappingMementoImpl( registrationName, resultMementos, fetchBuilderLegacies );
```

Comment:
Shouldn't this be handled by `EntityResultDescriptor#resolve` where the join descriptors are resolved into fetch mementos?